### DISCUSSION TITLE
FRDS + Meshtastic integration.

### DISCUSSION THREAD

**Discussion by dzuddiyn** [timmbogner/Farm-Data-Relay-System#224] [Ideas]
Hi,
what if we combine both FRDS (LoRa basic & ESP-now) with Meshtastic as  long haul repeater
AND as a messaging app for farmers & workers. in huge and rural farm, it's hard to get cellular network, so we can connect to Meshtastic WiFi to send message to  another worker in the other side of far-away farm plot. The Meshtastic WiFi should be with Good Strong antenna.
![photo_2024-06-28_15-28-39](https://github.com/timmbogner/Farm-Data-Relay-System/assets/173611048/693d6d42-a0e1-4d9e-aaa0-ec8560aa4e28)

end-nodes (LoRa / ESP-now) send to ->
the repeater send serial to ->
Meshtastic module.
Meshtasctic do the Mesh to far-away other Meshtastic.
at Main Hub House, a Meshtastic is connected to the Gateway by serial
& Gateway do its job.

Meshtastic is easy, just buy compatible module and flash. 
We could build it up with Solar & Good Batteries as Disaster Radio project did (https://hackaday.io/project/170069/gallery#b7ce3a826d8b701daea66c955bb18428)
![image](https://github.com/timmbogner/Farm-Data-Relay-System/assets/173611048/44dbb266-49c7-4a9a-98b8-6e1f0e4c9e1c) 
![image](https://github.com/timmbogner/Farm-Data-Relay-System/assets/173611048/1cbb23f6-1d5e-4400-bc1b-e9e059900d87)

Just my Idea...

**Comment by timmbogner** [comment]
It would probably be a fun project. Meshtastic generally uses the same LoRa hardware as FDRS so you won't have any range advantage, but you may find better flexibility in the topologies that are supported. I don't think Meshtastic LoRa packets will interfere with FDRS LoRa or vice versa. Let me how it works out!

**Comment by aviateur17** [comment]
I think there'd need to be some code changes so that the serial data from FDRS is compatible with what Meshtastic uses (I've never looked at Meshtastic myself).  Maybe that is the potential future ask.

**Comment by dzuddiyn** [comment]
I just watched the official Meshtastic serial function video. It talks about transparent serial mode and a bunch of other serial modes. It seems pretty easy to follow, maybe just follow to the data scheme in the documentation to get it.
if my project is approved, I will give a try. & update.

https://youtu.be/HdOiGKBtapw?si=hP6xyUUUgILQ3uGq

...

**Comment by aviateur17** [comment]
@dzuddiyn, looking at that video, if you set the Meshtastic device serial interface into Text mode on both sides I think you will get out exactly what you send in and so there would not need to be any modifications to FDRS.

**Reply by dzuddiyn** [reply]
oh thanks for the info.
I would try !!
